sim/src/mechanics/intersection.rs

```diff
@@ -766,6 +775,36 @@ impl IntersectionSimState {
     }
 }
 
+impl SignalState {
```

Comment:
This should hopefully be similar to the direction @khuston is taking, splitting into separate state.

Reply:
Yes. I also have `PhaseState`, which has properties that replace `stage_ends_at`.